src/js/remoteServices.js

```diff
@@ -115,6 +115,36 @@ export default {
     allowHostOverride: false,
   },
 
+  openproject: {
+    name: "openproject",
+    host: "",
+    urlPatterns: [
+      ":host:/projects/:project/work_packages/:id(/*)",
+      ":host:/projects/:project/work_packages/details/:id(/*)",
+      ":host:/work_packages/:id(/*)",
+      ":host:/work_packages/details/:id(/*)",
+    ],
+    description: (document) => {
+      let subject   = document.querySelector(".work-packages--details--subject")?.textContent?.trim()     || "";
+      let subjectId = document.querySelector(".work-packages--info-row")?.firstChild?.textContent?.trim() || "";
+
+      if (subjectId) {
+        subjectId = "OP " + subjectId;
+      }
+      if (subject && subjectId) {
+        subject = subjectId + " " + subject;
+      }
+
+      return subject || subjectId;
+    },
+    projectId: (document) => {
+      // ":project" in URL can be project name or OP internal project ID. Therefore, it cannot be used.
+      return (document.querySelector(".-project-context a") || document.querySelector("#projects-menu"))?.textContent?.trim() || "";
+    },
```

Comment:
The `projectId` should return the MOCO project number. The project number should be added to the OpenProject project name in square brackets, e.g. `Demo project [P123456]`, where P123456 is the project number in MOCO, and be extracted from there. You can use the `projectRegex` for that, see the other examples.

Reply:
I you would like to find a project by label, you might add an additional property `projectLabel` to the service. Square brackets in the project name should be removed, i.e. `Demo project [P1234566]` should become projectLabel `Demo project`

Reply:
Nice idea. I will add a new property `projectLabel`.